subprojects/docs/src/docs/userguide/troubleshooting/validation_problems.adoc

```diff
@@ -228,7 +228,11 @@ If you actually declared a file collection as an input in order to infer task de
 
 This error indicates that you have a task which depends on another, but that no explicit or implicit dependency is declared between those two tasks.
 As a consequence, the results of the build are dependent on the order of execution of tasks, often referred to "accidental dependencies between tasks".
-Often, this is because you refer directly to the output file of another task instead of using the task directly as an input.
+
+The solution is to add an <<more_about_tasks.adoc#sec:ordering_tasks,ordering>> or a <<more_about_tasks.adoc#sec:adding_dependencies_to_tasks,dependency>> between the two tasks.
```

Comment:
I wouldn't add this solution here: it should be done after the description of the problem (after "Gradle reports a missing dependency"). The risk here is that people click on the first link to figure out how to introduce ordering without having an understanding of the problem. We should only link when we have described to what use case the solution applies IMO.

Reply:
I moved it to the end of the section. How about now?